### PR TITLE
Add continuous delivery for plugin

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
We prepare the artifacts for release and we upload to the release bucket instead. Also we upload in parallel the artifacts as a Github Release.

#### Ticket Link
Ticket: https://mattermost.atlassian.net/browse/CLD-4681

### CODE REVIEW

File: .github/workflows/plugin-cd.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+on:
+  workflow_call:
+    inputs:
+      artifacts:
+        default: dist/*.tar.gz
+        description: |
+          The path or files of artifacts to upload
+        required: false
+        type: string
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:10]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      bucket:
+        default: mattermost-plugins-ci/release/
+        description: |
+          The S3 bucket full path to upload
+        required: false
+        type: string
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:17]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  release:
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - name: ci/download-artifact
+        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+        with:
+          name: dist
+          path: dist
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:45]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      - name: ci/prepare-artifact
+        run: |
+          mv *.tar.gz ${GITHUB_REPOSITORY#*/}-latest.tar.gz
+          cp ${GITHUB_REPOSITORY#*/}-latest.tar.gz ${GITHUB_REPOSITORY#*/}-$GITHUB_REF_NAME.tar.gz
+        working-directory: dist
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:51]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      - name: ci/aws-configure
+        uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # v1.7.0
+        with:
+          aws-region: ${{ inputs.region }}
+          aws-access-key-id: ${{ secrets.PLUGIN_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.PLUGIN_AWS_SECRET_ACCESS_KEY }}
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:58]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      - name: ci/artifact-upload
+        shell: bash
+        run: |
+          aws s3 cp ${{ inputs.artifacts }} s3://${{ inputs.bucket }} --acl public-read --cache-control no-cache
+        working-directory: dist
+
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:64]: Personal preference,  I would prefer loosing the NL here, for readability

File: .github/workflows/plugin-cd.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      - name: ci/publish-release
+        run: |
+          ghr -b "$(< ./release-notes.md)" -c ${GITHUB_REF_NAME} -n ${GITHUB_REF_NAME} -delete ${GITHUB_REF_NAME} *.tar.gz
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:67]: Better using the official `gh` tool for this
https://github.com/cli/cli

File: .github/workflows/plugin-cd.yml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        description: |
+          The Github token to publish artifacts
+        required: true
+
+permissions:
+  contents: read
+
+jobs:
+  release:
+    if: startsWith(github.ref, 'refs/tags')
```
Comment (phoinixgrr) [inline .github/workflows/plugin-cd.yml:37]:
```suggestion
    if: startsWith(github.ref, 'refs/tags/v')
```

Only release on tags starting with "v"

File: .github/workflows/plugin-cd.yml (hunk 7 above)
Comment (spirosoik) [reply .github/workflows/plugin-cd.yml]: @phoinixgrr fixed.

File: .github/workflows/plugin-cd.yml (hunk 2 above)
Comment (spirosoik) [reply .github/workflows/plugin-cd.yml:16]: on this section I like it which makes me easier to follow but overall 0/5 too.

**Comment by spirosoik** [comment]
@phoinixgrr have a look

**Review by phoinixgrr — APPROVED** [review]

**Pull request merged**